File: service-template-java/src/main/java/app/alphahunter/backend/site/constant/BusinessCode.java
package app.alphahunter.backend.site.constant;

public enum BusinessCode {
    SESSION_EXPIRED(-3, "session expired"),
    UNKNOWN_EXCEPTION(-2, "unknown exception"),
    COMMON_BIZ_ERROR(-1, ""),
    SUCCESS(0, "success");

    private final int code;
    private final String message;

    BusinessCode(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }
}

File: service-template-go/internal/app/constant/business_code.go
package constant

type BusinessCode struct {
	Code    int
	Message string
}

var (
	INVALID_REQUEST   = BusinessCode{-5, "invalid request"}
	SYSTEM_BUSY       = BusinessCode{-4, "system busy"}
	SESSION_EXPIRED   = BusinessCode{-3, "session expired"}
	UNKNOWN_EXCEPTION = BusinessCode{-2, "unknown exception"}
	COMMON_BIZ_ERROR  = BusinessCode{-1, "common biz error"}
	SUCCESS           = BusinessCode{0, "success"}
)
